test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -1997,3 +1997,23 @@ def test_basis_rotation_hermitian():
         Hadamard([2]),
         *hermitian.diagonalizing_gates(),
     ]
+
+
+def test_measurement():
+    qasm = """
+    bit[1] b;
+    qubit[2] q;
+    h q[0];
+    h q[1];
+    b[0] = measure q[0];
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    assert circuit.measured_qubits == [0]
+
+    qasm = """
+    bit[3] b;
+    qubit[3] q;
+    b = measure q;
+    """
+    circuit = Interpreter().build_circuit(qasm)
+    assert circuit.measured_qubits == [0, 1, 2]
```

Comment:
Can we make sure we're also testing:
- `measure q[0:1];`
- multiple measure statements (with and without overlap)
- multiple qubits/qubit registers defined
- `measure q[{0, 2}];`
- - `measure q[::-1];`
- remeasuring a qubit (should probably raise an error or be a noop, not sure yet)

Reply:
I will also add tests to validate the error message when an invalid qubit is measured (not in the range of qubits on the circuit or not a register)